Sources/Workspace/Workspace.swift

```diff
@@ -1520,55 +1503,65 @@ extension Workspace {
         }
 
         guard !diagnostics.hasErrors else {
-            return
+            throw Diagnostics.fatalError
         }
 
-        try self.download(artifactsToDownload, diagnostics: diagnostics)
+        // Download the artifacts
```

Comment:
👀 key changes here. instead of pre-defined artifact name in "subpath", the code now downloads the zip file, extract the content and checks for the artifact name. this allows us to not assume anything about the type of the file and instead base it on the file extensions. currently only xcframework file extension is supported, but its would be trivial to add other types

Reply:
This is great.  The .zip should really be just a container, and it's great that we're getting that.